From 19dc4a7461819d7eb99abd8a29abfd0d98232766 Mon Sep 17 00:00:00 2001
From: "Andrei V. FOMITCHEV" <andreix.fomitchev@intel.com>
Date: Fri, 8 Jun 2012 17:55:08 +0200
Subject: [PATCH 052/170] [PORT FROM MAIN] [Porting to JB][PreProc]accoustics
 effects process

BZ: 31726

The purpose of this patch is to propagate the errors of WebRTC to upper layers

Change-Id: I87c6812b621768f04b4ad80a38fa5604ce97c7ed
Origin-Change-Id: I32781aad02c55a9077241b8ada0bb6c28ace737f
Signed-off-by: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Reviewed-on: http://android.intel.com:8080/76840
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libeffects/preprocessing/PreProcessing.cpp |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/media/libeffects/preprocessing/PreProcessing.cpp b/media/libeffects/preprocessing/PreProcessing.cpp
index 597866a..c08de67 100755
--- a/media/libeffects/preprocessing/PreProcessing.cpp
+++ b/media/libeffects/preprocessing/PreProcessing.cpp
@@ -1281,7 +1281,7 @@ int PreProcessingFx_Process(effect_handle_t     self,
         session->procFrame->_payloadDataLengthInSamples =
                 session->apmFrameCount * session->inChannelCount;
 
-        effect->session->apm->ProcessStream(session->procFrame);
+        status = effect->session->apm->ProcessStream(session->procFrame);
 
         if (session->outBufSize < session->framesOut + session->frameCount) {
             session->outBufSize = session->framesOut + session->frameCount;
@@ -1325,11 +1325,10 @@ int PreProcessingFx_Process(effect_handle_t     self,
               (session->framesOut - fr) * session->outChannelCount * sizeof(int16_t));
         session->framesOut -= fr;
         outBuffer->frameCount += fr;
-
-        return 0;
     } else {
-        return -ENODATA;
+        status = -ENODATA;
     }
+    return status;
 }
 
 int PreProcessingFx_Command(effect_handle_t  self,
@@ -1791,11 +1790,11 @@ int PreProcessingFx_ProcessReverse(effect_handle_t     self,
         }
         session->revFrame->_payloadDataLengthInSamples =
                 session->apmFrameCount * session->inChannelCount;
-        effect->session->apm->AnalyzeReverseStream(session->revFrame);
-        return 0;
+        status = effect->session->apm->AnalyzeReverseStream(session->revFrame);
     } else {
-        return -ENODATA;
+        status = -ENODATA;
     }
+    return status;
 }
 
 
-- 
1.7.9

